ReactiveXComponentTest/WebSocket/WebSocketTests.cs

```diff
@@ -249,5 +249,84 @@ public void SnapshotTest()
+            {
+                var message = (string) callInfo.Args()[0];
+                var webSocketMessage = WebSocketMessageHelper.DeserializeRequest(message);
+                if (webSocketMessage.Command == WebSocketCommand.GetXCApi)
+                {
+                    var xcApiProperties = JsonConvert.DeserializeObject<WebSocketXCApiRequest>(webSocketMessage.Json);
+                    if(xcApiProperties.RequestId == requestId)
+                    {
+                        expectedRequestProperty = xcApiProperties.Name;
+                        requestSentEvent.Set();
+                    }
+
+                    var data = "getXcApi {\"RequestId\": \""+requestId+"\",\"ApiFound\":true,\"ApiName\":\"HelloWorld.api\",\"Content\":" + compressedXCApi +"}";
+                    var rawData = Encoding.UTF8.GetBytes(data);
+                    var messageEventArgs = new WebSocketMessageEventArgs(data, rawData);
+                    webSocketClient.MessageReceived += Raise.EventWith(messageEventArgs);
+                }
+            });
+
+            var xcApiManager = new WebSocketXCApiManager(webSocketClient);
+            var answer = xcApiManager.GetXCApi("HelloWorld.api", requestId);
+            var requestWasSent = requestSentEvent.WaitOne(TimeSpan.FromSeconds(5));
+
+            Check.That(requestWasSent).IsTrue();
+            Check.That(expectedRequestProperty.Equals("HelloWorld.api")).IsTrue();
+            Check.That(answer.Equals(expectedAnswer));
+        }
+
+        [Test]
+        public void GetXCApiListTest()
+        {
+            var expectedAnswer = new List<string> {"helloWorld.api"};
+            var requestId = Guid.NewGuid().ToString();
+            var requestSentEvent = new AutoResetEvent(false);
+            var webSocketClient = Substitute.For<IWebSocketClient>();
+            webSocketClient.IsOpen.Returns(true);
+            webSocketClient.When(x => x.Open()).DoNotCallBase();
+            webSocketClient.When(x => x.Close()).DoNotCallBase();
+            webSocketClient.When(x => x.Dispose()).DoNotCallBase();
+            webSocketClient.WhenForAnyArgs(x => x.Send("")).Do(callInfo =>
+            {
+                var message = (string) callInfo.Args()[0];
+                var webSocketMessage = WebSocketMessageHelper.DeserializeRequest(message);
+                if (webSocketMessage.Command == WebSocketCommand.GetXCApiList)
+                {
+                    requestSentEvent.Set();
+
+                    var data = "getXcApiList {\"RequestId\": \""+requestId+"\",\"Apis\":[\"helloWorld.api\"]}";
+                    var rawData = Encoding.UTF8.GetBytes(data);
+                    var messageEventArgs = new WebSocketMessageEventArgs(data, rawData);
+                    webSocketClient.MessageReceived += Raise.EventWith(messageEventArgs);
+                }
+            });
+
+            var xcApiManager = new WebSocketXCApiManager(webSocketClient);

```

Comment:
I can't: when creating a session a real websocketClient is created; and creating a session takes as argument a websocketEndpoint and opens websocketClient with this info. Unless using a real websocket connection, I can't use a websocketSession instance.